### ISSUE TITLE
Add an optional brand name/product identifier to sponsor and unpaid promotion segments

### ISSUE THREAD

**Issue by erkinalp** [ajayyy/SponsorBlock#1390]

**Comment by mchangrh** [comment]
Similar to the bias list discussed in the discord that would track which influences might be present across a video or channel

**Comment by ajayyy** [comment]
I'm worried that doing this could have unintended consequences

Some of them:

- Giving clients of the API an easy way to get payed off to whitelist specific segments
- Corporate analytics (brands using data to form their marketing strategy)

**Comment by erkinalp** [comment]
@ajayyy Yes that is true but it will allow us to skip spoken sponsor disclosures in favour of written disclosures, in videos with bias.

**Comment by mizlan** [comment]
An (unwieldy?) alternative might be to allow users to highlight a specific frame (or smaller segment) in the sponsor segment, where either the product name is shown onscreen or said aloud. This would allow viewers to quickly glance by the product being sponsored so they can be aware of it if it reappears elsewhere in the video.

**Comment by Pietagorh** [comment]
i feel like this would have the additional benefit of getting accurate data on youtube sponsors for research and journalism
i've seen a few people over the years talking about a given sponsor and guesstimating how "big" the ad campaign was
